node-services/cards-external-diffusion/src/domain/application/dailyCardsDiffusionControl.ts

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * See AUTHORS.txt
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of the OperatorFabric project.
+ */
+
+import CardsRoutingUtilities from './cardRoutingUtilities';
+import ConfigDTO from '../client-side/configDTO';
+import CardsDiffusionControl from './cardsDiffusionControl';
+import { log } from 'console';
+
+export default class DailyCardsDiffusionControl extends CardsDiffusionControl {
+    opfabUrlInMailContent: any;
+
+    protected dailyEmailTitle: string;
+    protected titlePrefix: string;
+
+
+    public setDailyEmailTitle(dailyEmailTitle: string) {
+        this.dailyEmailTitle = dailyEmailTitle;
+        return this;
+    }
+
+    public setConfiguration(updated: ConfigDTO) {
+        this.from = updated.mailFrom;
+        this.dailyEmailTitle = updated.dailyEmailTitle;
+    }
+
+
+    public async checkCardsOfTheDay() {
+        const users = this.cardsExternalDiffusionOpfabServicesInterface.getUsers();
+        const userLogins = users.map((u) => u.login);
+
+        const dateFrom = Date.now() - 24 * 60 * 60 * 1000;
+        const cards = await this.cardsExternalDiffusionDatabaseService.getCards(dateFrom);
+        userLogins.forEach(async (login) => {
+            const resp = await this.cardsExternalDiffusionOpfabServicesInterface.getUserWithPerimetersByLogin(login);
```

Comment:
Add a try catch for all the code in the foreach to avoid stopping if the process fails for one user